File: NOVEMBER-2025/14-11-2025/p2536.dart
class Solution {
  List<List<int>> rangeAddQueries(int n, List<List<int>> queries) {
    List<List<int>> mat = List.generate(
      n,
      (index) => List.filled(n, 0),
    );
    for (var q in queries) {
      int r1 = q[0], c1 = q[1], r2 = q[2], c2 = q[3];
      for (var i = r1; i <= r2; i++) {
        for (var j = c1; j <= c2; j++) {
          mat[i][j] = mat[i][j] + 1;
        }
      }
    }

    return mat;
  }
}

File: NOVEMBER-2025/14-11-2025/p2536.c
int** rangeAddQueries(int n, int** queries, int queriesSize,
                      int* queriesColSize, int* returnSize,
                      int** returnColumnSizes) {
    int diff[n + 1][n + 1] = {};
    for (int k = 0; k < queriesSize; k++) {
        int row1 = queries[k][0], col1 = queries[k][1], row2 = queries[k][2],
            col2 = queries[k][3];
        diff[row1][col1] += 1;
        diff[row2 + 1][col1] -= 1;
        diff[row1][col2 + 1] -= 1;
        diff[row2 + 1][col2 + 1] += 1;
    }

    int** mat = (int**)malloc(n * sizeof(int*));
    *returnColumnSizes = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        mat[i] = (int*)calloc(n, sizeof(int));
        (*returnColumnSizes)[i] = n;
    }
    *returnSize = n;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            int x1 = (i == 0) ? 0 : mat[i - 1][j];
            int x2 = (j == 0) ? 0 : mat[i][j - 1];
            int x3 = (i == 0 || j == 0) ? 0 : mat[i - 1][j - 1];
            mat[i][j] = diff[i][j] + x1 + x2 - x3;
        }
    }
    return mat;
}